client/MainWindow.cpp

```diff
@@ -50,6 +50,7 @@ MainWindow::MainWindow( QWidget *parent )
 	: QWidget( parent )
 	, cardsGroup( new QActionGroup( this ) )
 	, quitOnClose( false )
+	, doNotShowMsg( false )
```

Comment:
The variable name could be a little more descriptive and maybe instead of the negative "doNot" go with "do".
Something like "warnOnCancelofUnsignedDoc" or "warnOnUnsignedDocCancel"